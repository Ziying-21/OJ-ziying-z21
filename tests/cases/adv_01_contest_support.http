###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user7"}
###
# Got response:
# HTTP 200
# content-length: 23
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"id":1,"name":"user1"}
###
# Send request:
POST http://127.0.0.1:12345/contests HTTP/1.1
content-type: application/json

{"from":"2000-01-01T00:00:00.000Z","name":"Contest 1","problem_ids":[0],"submission_limit":1,"to":"2030-01-01T00:00:00.000Z","user_ids":[1]}
###
# Got response:
# HTTP 200
# content-length: 147
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"from":"2000-01-01T00:00:00.000Z","id":1,"name":"Contest 1","problem_ids":[0],"submission_limit":1,"to":"2030-01-01T00:00:00.000Z","user_ids":[1]}
###
# Send request:
POST http://127.0.0.1:12345/contests HTTP/1.1
content-type: application/json

{"from":"2000-01-01T00:00:00.000Z","name":"Contest 2","problem_ids":[1],"submission_limit":1,"to":"2030-01-01T00:00:00.000Z","user_ids":[0]}
###
# Got response:
# HTTP 200
# content-length: 147
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"from":"2000-01-01T00:00:00.000Z","id":2,"name":"Contest 2","problem_ids":[1],"submission_limit":1,"to":"2030-01-01T00:00:00.000Z","user_ids":[0]}
###
# Send request:
GET http://127.0.0.1:12345/contests HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 297
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# [{"from":"2000-01-01T00:00:00.000Z","id":1,"name":"Contest 1","problem_ids":[0],"submission_limit":1,"to":"2030-01-01T00:00:00.000Z","user_ids":[1]},{"from":"2000-01-01T00:00:00.000Z","id":2,"name":"Contest 2","problem_ids":[1],"submission_limit":1,"to":"2030-01-01T00:00:00.000Z","user_ids":[0]}]
###
# Send request:
GET http://127.0.0.1:12345/contests/1 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 147
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"from":"2000-01-01T00:00:00.000Z","id":1,"name":"Contest 1","problem_ids":[0],"submission_limit":1,"to":"2030-01-01T00:00:00.000Z","user_ids":[1]}
###
# Send request:
GET http://127.0.0.1:12345/contests/3 HTTP/1.1


###
# Got response:
# HTTP 404
# content-length: 68
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"code":3,"message":"Contest 3 not found.","reason":"ERR_NOT_FOUND"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":2,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":1}
###
# Got response:
# HTTP 400
# content-length: 79
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"code":1,"message":"User not in the contest.","reason":"ERR_INVALID_ARGUMENT"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":2,"language":"Rust","problem_id":0,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0}
###
# Got response:
# HTTP 400
# content-length: 82
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"code":1,"message":"Problem not in the contest.","reason":"ERR_INVALID_ARGUMENT"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":2,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 1137
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:11 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0},{"id":2,"info":"","memory":0,"result":"Waiting","time":0},{"id":3,"info":"","memory":0,"result":"Waiting","time":0},{"id":4,"info":"","memory":0,"result":"Waiting","time":0},{"id":5,"info":"","memory":0,"result":"Waiting","time":0},{"id":6,"info":"","memory":0,"result":"Waiting","time":0},{"id":7,"info":"","memory":0,"result":"Waiting","time":0},{"id":8,"info":"","memory":0,"result":"Waiting","time":0},{"id":9,"info":"","memory":0,"result":"Waiting","time":0},{"id":10,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:24:11.926Z","id":0,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":2,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/0 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1187
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:12 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":703985},{"id":1,"info":"","memory":0,"result":"Accepted","time":280965},{"id":2,"info":"","memory":0,"result":"Accepted","time":1871},{"id":3,"info":"","memory":0,"result":"Accepted","time":2206},{"id":4,"info":"","memory":0,"result":"Accepted","time":1635},{"id":5,"info":"","memory":0,"result":"Accepted","time":1606},{"id":6,"info":"","memory":0,"result":"Accepted","time":1807},{"id":7,"info":"","memory":0,"result":"Accepted","time":1517},{"id":8,"info":"","memory":0,"result":"Running","time":1488},{"id":9,"info":"","memory":0,"result":"Waiting","time":0},{"id":10,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:24:11.926Z","id":0,"result":"Running","score":70.0,"state":"Running","submission":{"contest_id":2,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/0 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1223
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:13 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":703985},{"id":1,"info":"","memory":0,"result":"Accepted","time":280965},{"id":2,"info":"","memory":0,"result":"Accepted","time":1871},{"id":3,"info":"","memory":0,"result":"Accepted","time":2206},{"id":4,"info":"","memory":0,"result":"Accepted","time":1635},{"id":5,"info":"","memory":0,"result":"Accepted","time":1606},{"id":6,"info":"","memory":0,"result":"Accepted","time":1807},{"id":7,"info":"","memory":0,"result":"Accepted","time":1517},{"id":8,"info":"","memory":0,"result":"Accepted","time":1488},{"id":9,"info":"","memory":0,"result":"Accepted","time":2391},{"id":10,"info":"","memory":0,"result":"Accepted","time":1583}],"created_time":"2022-09-09T02:24:11.926Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":2,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0},"updated_time":"2022-09-09T02:24:12.941Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":2,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0}
###
# Got response:
# HTTP 400
# content-length: 74
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:13 GMT
# 
# {"code":4,"message":"Submission limit exceeded","reason":"ERR_RATE_LIMIT"}
###
# Send request:
GET http://127.0.0.1:12345/contests/1/ranklist HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 58
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:13 GMT
# 
# [{"rank":1,"scores":[0.0],"user":{"id":1,"name":"user1"}}]
###
# Send request:
GET http://127.0.0.1:12345/contests/2/ranklist HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 59
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# date: Fri, 09 Sep 2022 02:24:13 GMT
# 
# [{"rank":1,"scores":[100.0],"user":{"id":0,"name":"root"}}]
