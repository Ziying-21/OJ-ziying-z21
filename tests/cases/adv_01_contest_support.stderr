[2022-09-09T02:24:11Z INFO  actix_server::builder] Starting 8 workers
[2022-09-09T02:24:11Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET / HTTP/1.1" 404 0 "-" "-" 0.000240
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000605
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /contests HTTP/1.1" 200 147 "-" "-" 0.000322
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /contests HTTP/1.1" 200 147 "-" "-" 0.000652
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests HTTP/1.1" 200 297 "-" "-" 0.000096
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/1 HTTP/1.1" 200 147 "-" "-" 0.000136
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/3 HTTP/1.1" 404 68 "-" "-" 0.000119
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 400 79 "-" "-" 0.000107
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 400 82 "-" "-" 0.000088
[2022-09-09T02:24:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 1137 "-" "-" 0.000496
[2022-09-09T02:24:12Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/0 HTTP/1.1" 200 1187 "-" "-" 0.000243
[2022-09-09T02:24:13Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/0 HTTP/1.1" 200 1223 "-" "-" 0.000320
[2022-09-09T02:24:13Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 400 74 "-" "-" 0.000174
[2022-09-09T02:24:13Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/1/ranklist HTTP/1.1" 200 58 "-" "-" 0.000152
[2022-09-09T02:24:13Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/2/ranklist HTTP/1.1" 200 59 "-" "-" 0.000114
